### PR TITLE
Add job model

### PR DESCRIPTION
**Pull request by d2lam**
- Add job model 
- Use jobModel.create inside pipeline model to create the job 'main'
- Remove trigger, triggeredBy

### CODE REVIEW

File: lib/pipelinemodel.js [hunk 1]
```diff
@@ -65,24 +66,13 @@ const PipelineModel = class {
                 return callback(error);
             }
 
-            // TODO: use job model
-            const name = 'main';
-            const id = hashr.sha1(`${pipelineId}${name}`);
+            const jobModel = new JobModel(this.datastore);
```
Comment (nkatzman) [inline lib/pipelinemodel.js:69]: would it be better to instantiate this in the configure?

Comment (nkatzman) [inline lib/pipelinemodel.js:69]: we can do it later if we need to

**Comment by nkatzman** [comment]
👍  once you add a README

**Comment by FenrirUnbound** [comment]
The README is a bit confusing. The examples include a HapiJS-specific implementation. Can we just make it straight to the point and show the model being used without the context fluff?

File: README.md [hunk 2]
```diff
@@ -98,6 +91,77 @@ sync(config, callback)
...
+#### Get
+Get a job based on id
+```
+get(id, callback)
+```
+
+| Parameter        | Type  |  Description |
+| :-------------   | :---- | :-------------|
+| id | String | The unique ID for the job |
+| callback | Function | Callback function|
```
Comment (FenrirUnbound) [inline README.md]: The callback isn't just a notification. The callback also returns back data. We should definitely document what gets passed to the callback function. Should we document what the return data looks like?

File: README.md [hunk 3]
```diff
@@ -98,6 +91,77 @@ sync(config, callback)
...
+```
+list(paginate, callback)
+```
+
+| Parameter        | Type  |  Description |
+| :-------------   | :---- | :-------------|
+| paginate        | Object | Pagination Object |
+| paginate.page | Number | The page for pagination |
+| paginate.count | Number | The count for pagination |
+| callback | Function | Callback function |
```
Comment (FenrirUnbound) [inline README.md]: Same concern with the returned data as the `get()`

File: lib/jobmodel.js [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+     */
+    constructor(datastore) {
+        this.datastore = datastore;
+        this.table = 'jobs';
+    }
+
+    /**
+     * Get a job based on id
+     * @param  {String}   id                The key of the record to retrieve
+     * @return {Function} callback
```
Comment (FenrirUnbound) [inline lib/jobmodel.js]: What does the method signature look like for the callback?

File: lib/jobmodel.js [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+
+        return this.datastore.get(config, callback);
+    }
+
+    /**
+     * List jobs with pagination
+     * @param  {Object}   paginate                  Config object
+     * @param  {Number}   paginate.count     Number of items per page
+     * @param  {Number}   paginate.page      Specific page of the set to return
+     * @return {Function} callback
```
Comment (FenrirUnbound) [inline lib/jobmodel.js]: What does the method signature look like for the callback?

File: lib/jobmodel.js [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+
+        return this.datastore.scan(config, callback);
+    }
+
+    /**
+     * Update a job
+     * @param  {Object}    config         Config object
+     * @param  {String}    config.id      The key of the record to retrieve
+     * @param  {Object}    config.data    The new data object to update with
+     * @return {Function}  callback
```
Comment (FenrirUnbound) [inline lib/jobmodel.js]: What does the method signature look like for the callback?

File: lib/jobmodel.js [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+            }
+        };
+
+        return this.datastore.scan(config, callback);
+    }
+
+    /**
+     * Update a job
+     * @param  {Object}    config         Config object
+     * @param  {String}    config.id      The key of the record to retrieve
```
Comment (FenrirUnbound) [inline lib/jobmodel.js]: What's the "key"?

File: test/lib/jobmodel.test.js [hunk 8]
```diff
@@ -0,0 +1,157 @@
...
+        datastore = null;
+        mockery.deregisterAll();
+        mockery.resetCache();
+    });
+
+    after(() => {
+        mockery.disable();
+    });
+
+    it('constructs', () => {
```
Comment (FenrirUnbound) [inline test/lib/jobmodel.test.js]: The other tests infer that the model can be properly created. Do we need this specific test?

File: test/lib/jobmodel.test.js [hunk 9]
```diff
@@ -0,0 +1,157 @@
...
+        const result = new JobModel(datastore);
+
+        assert.isOk(result);
+    });
+
+    it('has the correct API', () => {
```
Comment (FenrirUnbound) [inline test/lib/jobmodel.test.js]: The other tests infer that the public methods exist in order to assert their functionality. Do we need this specific test?

**Pull request merged**